packages/menu/src/MenuItem.ts

```diff
@@ -343,10 +360,10 @@ export class MenuItem extends LikeAnchor(
     }
 
     protected manageSubmenu(event: Event & { target: HTMLSlotElement }): void {
-        const assignedElements = event.target.assignedElements({
+        this.submenuElement = event.target.assignedElements({
             flatten: true,
-        });
-        this.hasSubmenu = !!assignedElements.length;
+        })[0] as HTMLElement;
```

Comment:
Can we keep a null pointer check here?

Reply:
The [return value](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSlotElement/assignedElements#return_value) for `assignedElements()` is always an array, even if it's empty, which means that accessing the `0` index will either get an `Element` or `undefined`, which are the two values that we expect here. Is there an alternative form of protection that you'd like to see a null pointer check guard us from?